batch/batch/server/server.py

```diff
@@ -158,65 +151,115 @@ def _has_next_task(self):
+                labels={'app': 'batch-job',
+                        'hail.is/batch-instance': INSTANCE_ID,
+                        'uuid': uuid.uuid4().hex
+                        }),
+            spec=pod_spec)
 
-        pod = v1.create_namespaced_pod(
-            HAIL_POD_NAMESPACE,
-            self._current_task.pod_template,
-            _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
-        self._pod_name = pod.metadata.name
+        try:
+            pod = v1.create_namespaced_pod(
+                HAIL_POD_NAMESPACE,
+                pod_template,
+                _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
+            self._pod_name = pod.metadata.name
 
-        await db.jobs.update_record(self.id,
-                                    pod_name=self._pod_name)
+            await db.jobs.update_record(self.id,
+                                        pod_name=self._pod_name)
 
-        log.info('created pod name: {} for job {}, task {}'.format(self._pod_name,
-                                                                   self.id,
-                                                                   self._current_task.name))
+            log.info('created pod name: {} for job {}, task {}'.format(self._pod_name,
+                                                                       self.id,
+                                                                       self._current_task.name))
+        except kube.client.rest.ApiException as err:
+            await self.mark_complete(None, failed=True)
+            log.info(f'pod creation failed for job {self.id} with the following error: {err}')
 
     async def _delete_pvc(self):
-        if self._pvc is not None:
-            log.info(f'deleting persistent volume claim {self._pvc.metadata.name} in '
-                     f'{self._pvc.metadata.namespace}')
+        if self._pvc_name is not None:
+            log.info(f'deleting persistent volume claim {self._pvc_name}')
             try:
                 v1.delete_namespaced_persistent_volume_claim(
-                    self._pvc.metadata.name,
+                    self._pvc_name,
                     HAIL_POD_NAMESPACE,
                     _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
             except kube.client.rest.ApiException as err:
                 if err.status == 404:
-                    log.info(f'persistent volume claim {self._pvc.metadata.name} in '
-                             f'{self._pvc.metadata.namespace} is already deleted')
+                    log.info(f'persistent volume claim {self._pvc_name} is already deleted')
                     return
                 raise
             finally:
-                await db.jobs.update_record(self.id, pvc=None)
+                await db.jobs.update_record(self.id, pvc_name=None)
+                self._pvc_name = None
 
     async def _delete_k8s_resources(self):
         await self._delete_pvc()
+        await db.jobs.update_record(self.id, pod_name=None)
```

Comment:
i think we should be consistent about updating before or in a finally block.why is this one different from _delete_pvc?

Reply:
I agree it should be consistent. What is your preference? I don't know what the right answer should be. Just thought it was better to set it to null in the db early in case something failed.

Reply:
I don't think it matters.  The finally should still run even if something fails.